R/GitHost.R

```diff
@@ -0,0 +1,131 @@
+#' @importFrom R6 R6Class
+#' @importFrom rlang expr
+#' @importFrom cli cli_alert_danger cli_alert_success
+
+#' @title A GitHost superclass
+
+GitHost <- R6::R6Class("GitHost",
+  public = list(
+
+    #' @description Create a new `GitPlatform` object
+    #' @param orgs A character vector of organisations (owners of repositories
+    #'   in case of GitHub and groups of projects in case of GitLab).
+    #' @param token
+    #' @param api_url
+    #' @param settings
+    #' @return A new `GitPlatform` object
+    initialize = function(orgs = NA,
```

Comment:
why do you rewrite settings to the Host object? I would just keep them inside Gitstats object and pass them down as an argument for get_commits() and get_repos() to avoid confusion and possible mismatches (when GitStats and GitHost might be having different settings that should be checked out at eachstep)